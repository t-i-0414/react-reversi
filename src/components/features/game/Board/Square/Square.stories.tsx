import { Meta, StoryObj } from '@storybook/react';
import { Square } from './Square';

const meta: Meta<typeof Square> = {
  title: 'features/game/Board/Square',
  component: Square,
  argTypes: {
    dataCy: { control: { type: 'text' } },
  },
  args: {
    dataCy: 'dataCy',
  },
};

export default meta;

type Story = StoryObj<typeof Square>;

export const Normal: Story = {
  render: args => <Square {...args} />,
};
