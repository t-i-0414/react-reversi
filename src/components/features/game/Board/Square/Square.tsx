import React, { memo } from 'react';
import styled from 'styled-components';
import { SizeMap, ColorMap } from '~/constants';

export interface SquareProps {
  dataCy?: string;
  children?: React.ReactNode;
}

export const Square: React.FC<SquareProps> = memo(({ children, dataCy }) => (
  <Wrapper data-cy={dataCy}>{children}</Wrapper>
));
Square.displayName = 'Square';

const Wrapper = styled.div`
  box-sizing: border-box;
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${SizeMap.SQUARE_SIZE}px;
  height: ${SizeMap.SQUARE_SIZE}px;
  background-color: ${ColorMap.BG_GREEN};
  border: 1px solid ${ColorMap.BD_BLACK};
`;
