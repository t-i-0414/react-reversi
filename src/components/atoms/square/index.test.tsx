import React from 'react';
import renderer from 'react-test-renderer';
import Square from '.';

describe('square unit test', () => {
  it('snapshot', () => {
    const component = renderer.create(<Square dataCy="square-1" />);
    expect(component).toMatchSnapshot();
  });
});
