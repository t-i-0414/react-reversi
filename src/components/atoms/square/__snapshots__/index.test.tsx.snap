// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`square unit test snapshot 1`] = `
<div
  className="sc-beySbM iuPTYZ"
  data-cy="square-1"
/>
`;
